Add skipHardware early return to yOCTScanGlassSlideToFindFocusAndDisp…

alejoruge opened the pull request:
…ersionQuadraticTerm

Add skipHardware early return to glass slide focus/dispersion function

Current situation & Problem

- yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm accepts skipHardware parameter but doesn't handle it
- Function attempts to read non-existent scan files when skipHardware=true
- Causes errors in simulation mode after yOCTScanTile returns early
- No way to skip focus/dispersion measurement during testings

Release Notes

- Added early return when skipHardware=true (lines 41-51)
- Returns NaN for both dispersionQuadraticTerm and focusPositionInImageZpix
- NaN values clearly indicate unmeasured parameters in simulation mode
- Prints informative message in verbose mode
- Prevents file read errors and unnecessary processing
- Consistent with skipHardware behavior in yOCTScanTile and other myOCT functions

alejoruge committed:
Add skipHardware early return to yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm

Add skipHardware early return to glass slide focus/dispersion function

Current situation & Problem

- yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm accepts skipHardware parameter but doesn't handle it
- Function attempts to read non-existent scan files when skipHardware=true
- Causes errors in simulation mode after yOCTScanTile returns early
- No way to skip focus/dispersion measurement during testings

Release Notes

- Added early return when skipHardware=true (lines 41-51)
- Returns NaN for both dispersionQuadraticTerm and focusPositionInImageZpix
- NaN values clearly indicate unmeasured parameters in simulation mode
- Prints informative message in verbose mode
- Prevents file read errors and unnecessary processing
- Consistent with skipHardware behavior in yOCTScanTile and other myOCT functions

MyYo reviewed (changes requested):
There is implementation of skipHardware in line 92.
Can you explain more about why you implement here?

My hope is that by using skipHardware, we can still debug the function though just not the hardware part. So if there is a way not to bypass it altogether I think its better.

Can you explain the scenario in which you experienced an issue?

alejoruge commented:
Oh I see, I didn't think about it that way. The problem comes at line 88 where it tries to read ScanInfo.json, but that file doesn't exist when skipHardware=true because yOCTScanTile doesn't create it.

I was thinking about an alternative: what if we check skipHardware right before line 88 and generate mock data only for that part? Would you like me to try that?

Right now, every time we have skipHardware = true, we need to comment the whole function out because it runs the error..

alejoruge committed:
Implement skipHardware mode without runtime errors

Add complete skipHardware support to yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm
to enable testing and debugging of calibration algorithms without hardware.

Changes:
- Use yOCTSimulateTileScan to generate complete mock OCT data with file structure
- Create simulation/ subdirectory within tempFolder to separate simulated from real data
- Get json directly from yOCTScanTile/yOCTSimulateTileScan return value instead of reading ScanInfo.json file
- Pass OCTSystem explicitly to yOCTLoadInterfFromFile in skipHardware mode to avoid detection errors with simulated data files
- Default atFocusIndex to middle scan in skipHardware mode when intensity-based detection fails (simulated data has uniform intensity across depths)
- Add double() conversion in alignZ function for fminsearch compatibility
- Add bounds clipping for maxIdZEnv indices to handle edge cases without affecting normal hardware operation
- Generate multi-layer reflector structure in simulated data (glass interface +
  secondary reflections) for realistic OCT signal processing

alejoruge committed:
Remove the early return to use real debugger

alejoruge committed:
Cleanup simulated data only

MyYo commented on Applications/yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm.m:
```diff
@@ -73,24 +83,44 @@
         for scanI = 1:length(json.octFolders)
-            [interf, dim] = yOCTLoadInterfFromFile(...
-                [tempFolder json.octFolders{scanI}], 'YFramesToProcess',1);
+            % For simulated data, pass OCTSystem explicitly to avoid detection error
```
No need for duplication here you can just use 
'''
[interf, dim] = yOCTLoadInterfFromFile([tempFolder json.octFolders{scanI}], ...
                    'YFramesToProcess', 1, ...
                    'OCTSystem', json.OCTSystem);
'''
For both cases

MyYo commented on Applications/yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm.m:
```diff
@@ -107,6 +137,13 @@
             end
         end
 
+        % In simulation mode, intensity scores may be uniform across all scans
```
I don't think this is needed. If all intensities have same value a valid atFocusIndex is still selected.

alejoruge committed:
Merge branch 'master' into simulated-focus-disp

# Conflicts:
#	Applications/yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm.m

alejoruge committed:
Add print standard format

MyYo reviewed (approved):
looks good.
Small comment

alejoruge committed:
Update

alejoruge committed:
Update yOCTScanGlassSlideToFindFocusAndDispersionQuadraticTerm.m

alejoruge committed:
Correct else comment
